lib/settings.h

```diff
@@ -167,6 +167,9 @@ class CPPCHECKLIB WARN_UNUSED Settings {
     /** cppcheck.cfg: About text */
     std::string cppcheckCfgAbout;
 
+    /** @brief check Emacs marker to detect extension-less and *.h files as C++ */
+    bool cppHeaderProbe{};
```

Comment:
I wonder why this is not on by default. Do you have an idea why people would want to disable it?

If `--language` is used would that override the emacs marker? I assume that would be nice?

Do I understand it correctly that the emacs marker is only read in the first file.. cppcheck will not try to identify included headers?

Reply:
> I wonder why this is not on by default. Do you have an idea why people would want to disable it?

Because it changes the behavior and is not completely implemented yet.

It should not have any impact though since you usually would not pass such files to the analysis.

I also want to use a bit via the IDE integration I use but for it to have effect we also need to merge #6346.

I wonder if we should add a (debug) message if the language could not be determined.

> If `--language` is used would that override the emacs marker? I assume that would be nice?

It does. That is done later on in the tokenizing via `enforcedLanguage` using that complicated logic which involves a lot of asserts. That should (hopefully) also simplify a lot after the usage of `Path::identify()` has been cleaned up. As mentioned before that needs more tests and such first.

> Do I understand it correctly that the emacs marker is only read in the first file.. cppcheck will not try to identify included headers?

Yes. The initial input determines how the compiler (and thus we) is supposed to handle it. If you compile a file as C++ everything included is obviously also treated as C++.

Reply:
> Because it changes the behavior and is not completely implemented yet.

When it is completely implemented.. I don't feel I would be against that this is enabled by default.

> It should not have any impact though since you usually would not pass such files to the analysis.

Yes I agree.

> I wonder if we should add a (debug) message if the language could not be determined.

Sounds OK to me.